### ISSUE TITLE
NX

### ISSUE THREAD

**Issue by Cr4xy** [mRemoteNG/mRemoteNG#355]
Please add the NX protocol for RDP, maybe you can use http://opennx.net/. Thanks!

**Comment by kmscode** [comment]
For reference:
https://www.nomachine.com
https://en.wikipedia.org/wiki/NX_technology

**Comment by Sonic3R** [comment]
Any progress of it ?

**Comment by merarischroeder** [comment]
In mRemoteNG, SSH works by opening Putty and then making the Tab-Container the parent of the Putty window. This same approach is likely possible with NoMachine.

NoMachine is Freeware, and they make money from the Server-Side. So mRemoteNG should be able to download the client on demand.

Feasibility Prototype is required

- Install NoMachine on your computer. Are there good command-line arguments so that you can open directly into a connection? [15-mins]
- Will the NoMachine window be contained? Modify the SSH code so that it starts NoMachine instead of Putty. Run the program, and create an SSH connection with IP, username, and password. Connect. Does it work? [2 hours]

If the above feasibility prototypes work, then it will be quite easy to add NX support. I suspect this is the same for other desired protocols as well.

**Comment by Kvarkas** [comment]
@merarischroeder we are using a bit modified putyNg as we not manage to use it directly, if you wish to play in that direction its more than welcome

**Comment by merarischroeder** [comment]
@Kvarkas 

Yes, I did notice PuttyNg when I was looking through the code. I did forget. That's a good point.

How does your "Try to Integrate" work for [External Tools](https://mremoteng.readthedocs.io/en/latest/user_interface/external_tools.html)? It it similar to the Parent Container approach?

I think External Tools is a really good capability to have. Even if it opens in a separate window, the Connection Management is still very useful (the tabs aren't essential and can even get in the way of multi-monitor).

**Comment by Kvarkas** [comment]
@merarischroeder unfortunately I have not worked on that bit so have very basic understanding about how its implemented :(

**Comment by merarischroeder** [comment]
`NativeMethods.SetParent(_handle, InterfaceControl.Handle);`

It would appear so. The above line is in IntegratedProgram.cs on line 101. `SetParent` is a direct PInvoke method in Windows API.

**Comment by Kvarkas** [comment]
OpenNX is a discontinued open-source client for remote desktop access, originally designed as a replacement for NoMachine's NX client. It is now largely obsolete, with limited use among legacy systems and niche developers.
